docs: landing and architecture page updates (#26)
armory/docs-cdaas: content/en/includes/cdaas-explained-how.md

> ## How Armory CD-as-a-Service works

Reviewer: ```suggestion
## How CD-as-a-Service works
```

> Armory CD-as-a-Service is a centralized control plane that utilizes flexible promotion constraints to safely rollout software/config changes across multiple clusters, environments, and/or regions.

Reviewer: ```suggestion
CD-as-a-Service is a centralized control plane that utilizes flexible promotion constraints to safely deploy software and/or config changes across multiple clusters, environments, and/or regions.
```

{{< figure src="/images/cdaas/cdaas-arch.png" alt="CD-as-a-Service High-Level Architecture" height="75%" width="75%" >}}

> CD-as-a-Service uses secure logic-less Remote Network Agents to access privately networked resources and Kubernetes clusters via Service Accounts, you do not need upgrade agents for new features.

Reviewer: ```suggestion
CD-as-a-Service uses secure, logicless _Remote Network Agents_ to access privately networked resources and Kubernetes clusters via ServiceAccounts. You do not need upgrade agents for new features since business logic is encapsulated in the CD-as-as-Service control plane.
```

Use italics to introduce a new term. Remote Network Agent is a new term. https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-italics-to-define-or-introduce-new-terms

> Armory's Remote Network Agents connect to CD-as-a-Service to establish gRPC via HTTP/2 connections secured with TLS and OIDC client credentials. This means that you don't need to open any ports to grant the Armory CDaaS control plane access to your Kubernetes clusters or privately networked resources.

Reviewer: ```suggestion
Armory's Remote Network Agents connect to CD-as-a-Service to establish gRPC via HTTP/2 connections secured with TLS and OIDC client credentials. You don't need to open any ports to grant the CD-as-a-Service access to your Kubernetes clusters or privately networked resources.
```

Use simple and direct language https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-simple-and-direct-language
You don't need to use phrases like "This means that..."

With CD-as-a-Service:

> 1. You build and publish your containers where and how you want, from dockerhub to a private registry that is only accessible in your network.

Reviewer: ```suggestion
1. You build and publish your containers where and how you want, whether you use  Docker Hub or a private registry that is only accessible in your network.
```

> 2. You render your kubernetes manifests how you want (Helm, Kustomize, Mustache, raw manifests, etc).

Reviewer: ```suggestion
2. You create your Kubernetes manifests how you want - manually or by using a tool like Helm or  Kustomize.
```

"render" isn't really the right word here -> you missed the discussion on this during Wed office hours.  Narrowed it down to Helm and Kustomize since Parth said those 2 are the most popular.

> 3. You define your software delivery requirements, traffic shaping, canary analysis, webhooks, manual approvals, etc, through our [declarative deployment configuration file](/deployment/overview).

Reviewer: ```suggestion
3. You define your software delivery requirements, traffic shaping, canary analysis, webhooks, and manual approvals in a [declarative deployment configuration file]({{< ref "deployment/overview.md" >}}).
```

Better to use `ref` or `linkWithTitle` since the build will fail if Hugo can't find those pages. Very useful when refactoring (such as changing file names for seo).

> Start your deployment by sending the deployment configuration file and rendered manifests to CD-as-a-Service using the [Armory CLI](/cli) or automatically with a CI integration like our [GitHub Action](/integrations/ci-systems/gh-action).

Reviewer: ```suggestion
Start your deployment by sending the deployment config file and rendered manifests to CD-as-a-Service using the [Armory CLI]({{< ref "cli" >}}) or automatically with a CI integration like CD-as-a-Service's [GitHub Action]({{< ref "integrations/ci-systems/gh-action" >}}).
```

Maintain term consistency - **deployment config file** is what the current docs set uses,. I settled on that specific phrase based on reader feedback on what they were searching for.

> The control plane executes the deployments with constraints defined in your deployment config, converting Kubernetes deployment objects into CD-as-a-Service managed ReplicaSets, handling traffic management, scaling and more. Remote Network Agents integrate with internal tools by securely relaying network calls and using their configurable service account credentials to communicate with your clusters API.

Reviewer: ```
The CD-as-a-Service control plane executes the deployments with constraints defined in your deployment config file, converting Kubernetes deployment objects into ReplicaSets managed by CD-as-a-Service, handling traffic management, scaling and more. Remote Network Agents integrate with internal tools by securely relaying network calls and using their configurable ServiceAccount credentials to communicate with your cluster's API.
```

`scaling, and more`: either add more to the list or remove `, and more`.

```
Remote Network Agents integrate with internal tools by securely relaying network calls and using **their** configurable ServiceAccount credentials to communicate with your cluster's API.
```
Does `their` refer to the RNAs or the internal tools? It's not clear to me based on the sentence structure.

Maintain term consistency - deployment config file is what the current docs set uses,. I settled on that specific phrase based on reader feedback on what they were searching for.
